libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -560,10 +560,12 @@
 #endif // _CCCL_CHECK_BUILTIN(underlying_type) && gcc >= 4.7
 
 #if defined(_CCCL_COMPILER_MSVC)
-#  if _CCCL_MSVC_VERSION >= 1935
+#  // To use __builtin_FUNCSIG(), both MSVC and nvcc need to support it
+#  if _CCCL_MSVC_VERSION >= 1935 && (!defined(_CCCL_CUDACC) || _CCCL_CUDACC_VER >= 1203000)
```

Comment:
I believe this should be should be 
```suggestion
#  if _CCCL_MSVC_VERSION >= 1935 && !defined(_CCCL_CUDACC_BELOW_12_3)
```

Reply:
I don't think so because of #2686 but I could be wrong. There are a lot of double negatives involved here.